src/Emitter.ts

```diff
@@ -164,70 +271,391 @@ export class Emitter
+
+            // IMemberDoclet:
+            case 'member':
+            case 'constant':
+                if (isDefaultExportDoclet(node.doclet, this._treeNodes))
+                {
+                    if (node.doclet.meta && node.doclet.meta.code.value)
+                    {
+                        this._resolveDocletType(node.doclet.meta.code.value, node, this._markExportedNode);
+                    }
+                }
+                else if (isNamedExportDoclet(node.doclet, this._treeNodes)
+                         && node.doclet.meta && node.doclet.meta.code.value
+                         && (! isEnumDoclet(node.doclet)))
+                {
+                    const thisEmitter = this;
+                    this._resolveDocletType(node.doclet.meta.code.value, node, function (refNode: IDocletTreeNode) {
+                            // Directly mark the referenced node for export (through this path at least), only if the exported name is changed.
+                            const markThisNode = node.doclet.meta && (node.doclet.meta.code.value === node.doclet.name);
+                            thisEmitter._markExportedNode(refNode, markThisNode);
+                        },
+                        // Doclet filter: avoid cyclic loops in case this named export references a type with the same name.
+                        function(target: IDocletTreeNode) {
+                            return (target !== node);
+                        }
+                    );
+                }
+                else
+                {
+                    this._markExportedTypes(node, node.doclet.type);
+                }
+                break;
+
+            // INamespaceDoclet:
+            case 'module':
+                // Search for export doclets in the module.
+                for (const child of node.children)
+                {
+                    if (isDefaultExportDoclet(child.doclet, this._treeNodes)
+                        || isNamedExportDoclet(child.doclet, this._treeNodes))
+                    {
+                        this._markExportedNode(child);
+                    }
+                }
+                break;
+            case 'namespace':
+                this._markExportedChildren(node);
+                break;
+
+            // ITypedefDoclet:
+            case 'typedef':
+                this._markExportedTypes(node, node.doclet.type);
+                // When the typedef is for a function, the doclet may have params and returns.
+                this._markExportedParams(node, node.doclet.params);
+                this._markExportedReturns(node, node.doclet.returns);
+                break;
+
+            default:
+                return assertNever(node.doclet);
+        }
+    }
+
+    private _markExportedTypes(node: IDocletTreeNode, types?: IDocletType)
+    {
+        if (types)
+            for (const typeName of types.names)
+                this._resolveDocletType(typeName, node, this._markExportedNode);
```

Comment:
Let's add some braces for all these helper functions that are missing them.

Reply:
Sorry, I'm not sure to get it.

Reply:
I mean:

```
        if (types)
        {
            for (const typeName of types.names)
            {
                this._resolveDocletType(typeName, node, this._markExportedNode);
            }
        }
```

Reply:
Got it.
Braces added in `_markExportedTypes(()`, `_markExportedParams()`, `_markExportedReturns()` and `_markExportedChildren(()`.